plugins/auth-backend/src/extensions.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createExtensionPoint } from '@backstage/backend-plugin-api';
+import { ProviderFactories } from '@backstage/plugin-auth-backend';
+
+/**
+ * @alpha
+ */
+export interface AuthProviderExtensionPoint {
+  addProviderFactories(factories: ProviderFactories): void;
+}
+
+/**
+ * @alpha
+ */
+export const authProviderExtensionPoint =
```

Comment:
These two actually need to live in a new `auth-node` package with `"role": "node-library"` - since they must be imported by other plugins and modules, and plugins shouldn't import plugins.

Reply:
I moved them, as well as  a nulber of types involved in AuthProvider definitions, to `auth-node`.
In order to avoid a very big number of import changes in the `auth-backend` code, I reexported them in the corresponding `types.ts` files in `auth-backend`.